src/builder.cpp

```diff
@@ -482,12 +487,15 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
 				continue;
 			}
 
-			// Add surface to mesh
-			add_surface_to_mesh(mesh, surf);
+			add_surface_to_mesh(collision_mesh, surf);
 
-			// Give mesh material
-			if (material != nullptr) {
-				mesh->surface_set_material(mesh->get_surface_count() - 1, material);
+			// Add surface to mesh
```

Comment:
Misplaced comment? These should ultimately have 2 different comments, one for adding the surface to the visible mesh, and one for adding the surface to the collision mesh.